frontend/src/pages/projects/screens/detail/notebooks/NotebookTable.tsx

```diff
@@ -34,9 +34,10 @@ const NotebookTable: React.FC<NotebookTableProps> = ({
             ))}
           </Tr>
         </Thead>
-        {sortedNotebookStates.map((notebookState) => (
+        {sortedNotebookStates.map((notebookState, i) => (
```

Comment:
This will have some adjustments inside the new Table.tsx component. 
the prop `rowRenderer` in `Table` will need to be changed to:
```tsx
rowRenderer: (data: DataType, rowIndex: number) => React.ReactNode;
```

However I am afraid that adding a row index messes up how react optimizes rendering lists of items. Because that row index will change every time a filter is added or pagination is used.